zerver/lib/actions.py

```diff
@@ -4717,15 +4718,24 @@ def do_delete_messages(realm: Realm, messages: Iterable[Message]) -> None:
         subscribers_ids = [user.user_profile_id for user in subscribers]
         users_to_notify = list(map(subscriber_info, subscribers_ids))
 
-    move_messages_to_archive(message_ids, realm=realm)
+    if message_type == "stream":
+        move_messages_to_archive(message_ids, realm=realm, chunk_size=retention.STREAM_MESSAGE_BATCH_SIZE)
+    else:
+        move_messages_to_archive(message_ids, realm=realm)
```

Comment:
I think this would read better if we just put the `if` statement in selecting a `chunk_size` value, and then had one call to `move_messages_to_archive`.

`do_delete_messages_by_sender` can probably just use `STREAM_MESSAGE_BATCH_SIZE` with a comment; I don't think it's worth the extra complexity to try to differentiate there.

Reply:
Agreed on both, fixed :slightly_smiling_face: